config/src/errors.rs

```diff
@@ -0,0 +1,180 @@
+//! Errors for graph-sdk crate
+//!
+
+use thiserror::Error;
+
+pub type DsnpGraphResult<T> = std::result::Result<T, DsnpGraphError>;
+
+#[derive(Debug, Error)]
+pub enum DsnpGraphError {
+	// Errors related to Avro
+	#[error("Avro error: {0}")]
+	AvroError(String),
```

Comment:
I sorted them but since they are not unit variants and contains tuples we cannot achieve pinning the values like =1 etc.
One way to achieve this especially for ffi is assign them some code via function call, 

Let me know how you feel about it

Reply:
Not necessarily true; for instance, given the following enum:
```
#[repr(C)]
pub enum MyError {
   Error1 = 1,
   Error2(u8) = 2,
   Error3(String) = 3,
   Error4 { uid: u8, name: String } = 4,
}
```

cbindgen does generate integer-valued C-style enums. Here is the cbindgen output for that enum:
```
typedef struct String String;

typedef enum {
  Error1 = 1,
  Error2 = 2,
  Error3 = 3,
  Error4 = 4,
} MyError_Tag;

typedef struct {
  uint8_t uid;
  String name;
} Error4_Body;

typedef struct {
  MyError_Tag tag;
  union {
    struct {
      uint8_t error2;
    };
    struct {
      String error3;
    };
    Error4_Body error4;
  };
} MyError;
```

See here for more explanation:
https://rust-lang.github.io/unsafe-code-guidelines/layout/enums.html#:~:text=Explicit%20repr%20annotation%20with%20C%20compatibility

It may not do any good in Rust, but the error struct exposed to the C API would expose the integer value of the enum.